core/src/main/java/io/questdb/cairo/wal/seq/TableMetadataChange.java

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.wal.seq;
+
+import io.questdb.cairo.wal.TableWriterSPI;
+import io.questdb.griffin.SqlException;
+
+@FunctionalInterface
+public interface TableMetadataChange {
+    // todo: should not be SqlException
+    long apply(TableWriterSPI tableWriterSPI, boolean flag) throws SqlException;
```

Comment:
nit: `boolean flag` is not the best name for a boolean argument. Should be `contextAllowsAnyStructureChanges`.

Reply:
renamed it and made `AsyncWriterCommand` extend this interface instead of duplicating the method.
it was a bit confusing, AlterOperation inherited it from both places.

Reply:
Thanks!